core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/controller/promisehandlers/MonitoringHandler.scala

```diff
@@ -34,11 +66,12 @@ trait MonitoringHandler {
 
       // send Monitoring message
       val requests = workers.map(worker =>
-        send(QuerySelfWorkloadMetrics(), worker).map(metric => {
+        send(QueryWorkloadMetrics(), worker).map(value => {
           workflow.getOperator(worker).getWorkerWorkloadInfo(worker).dataInputWorkload =
-            metric.unprocessedDataInputQueueSize + metric.stashedDataInputQueueSize
+            value._1.unprocessedDataInputQueueSize + value._1.stashedDataInputQueueSize
```

Comment:
the renaming to `value._1` make is unreadable. can we change the naming? for example use 
```
.map ({
    (name1, name2) => {func();}
})
```

Reply:
I tried doing this, but it seems like tuple2 isn't allowed inside scala lambda. I renamed `value` to `metricAndSample`, so that metricAndSample._1 gives an intuition about what the result will be.